src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_scans.rs

```diff
@@ -1117,50 +1103,52 @@ impl Scan {
                 break;
             }
 
-            self.concurrent_parse_bytes(state, &buf[..len]);
+            self.parse_bytes(state, &buf[..len]);
         }
 
         /* Scanned all blocks */
         Ok(())
     }
+}
+
+pub struct Scan {
+    engine: ScanEngine,
+    state: ScanState,
+}
+
+impl Scan {
+    pub fn new(options: ScanOptions) -> Self {
+        Self {
+            engine: ScanEngine::new(options),
+            state: ScanState::default(),
```

Comment:
@beaubelgrave This is a breaking change for existing users right? Now users need to access checks via the scan state object.
If my understanding is correct, then we need to add an entry in the release notes highlighting this. I can take care of the release note entry.

Reply:
@jdraaijer-microsoft is correct, this isn't breaking. However, it's nice to call out the ability to use this for scans while owning the state for concurrent designs.